MdePkg/Library/PeiHobLib/HobLib.c

```diff
@@ -342,6 +343,55 @@ BuildResourceDescriptorWithOwnerHob (
   CopyGuid (&Hob->Owner, OwnerGUID);
 }
 
+// MU_CHANGE Start: Add BuildResourceDescriptorV2 function
+/**
+  Builds a HOB that describes a chunk of system memory with memory attributes.
+
+  This function builds a HOB that describes a chunk of system memory.
+  It can only be invoked during PEI phase;
+  for DXE phase, it will ASSERT() since PEI HOB is read-only for DXE phase.
+
+  If there is no additional space for HOB creation, then ASSERT().
+
+  @param  ResourceType          The type of resource described by this HOB.
+  @param  ResourceAttribute     The resource attributes of the memory described by this HOB.
+  @param  PhysicalStart         The 64 bit physical address of memory described by this HOB.
+  @param  NumberOfBytes         The length of the memory described by this HOB in bytes.
+  @param  EfiMemoryAttributes   The memory attribute for the memory described by this HOB.
+

```

Comment:
`OwnerGUID` param description is missing.

Reply:
Added. Nice call.